catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/utils/BrickTestUtils.java

```diff
@@ -84,4 +98,36 @@ public static void createUserListFromDataFragment(String userListName, boolean f
 		onView(withText(userListName))
 				.check(matches(isDisplayed()));
 	}
+
+	public static void addBrick(Class<?> brickHeaderClass, int brickCategoryId) {
+		onView(withId(R.id.button_add))
+				.perform(click());
+		onData(allOf(Is.is(instanceOf(String.class)), Is.is(UiTestUtils.getResourcesString(brickCategoryId))))
+				.inAdapterView(BrickCategoryListMatchers.isBrickCategoryView())
+				.perform(click());
+		onData(Is.is(instanceOf(brickHeaderClass))).inAdapterView(BrickPrototypeListMatchers.isBrickPrototypeView())
+				.perform(click());
+
+		onData(instanceOf(Brick.class))
+				.inAdapterView(ScriptListMatchers.isScriptListView()).atPosition(1)
+				.perform(click());
+	}
+
+	public static void deleteItemWithTextFromFormulaEditorDataList(String text) {
+		onView(withId(R.id.formula_editor_keyboard_data))
+				.perform(click());
+		onView(allOf(withText(text), withId(R.id
+				.fragment_formula_editor_datalist_item_name_text_view)))
+				.perform(longClick());
+		onView(withText(R.string.delete))
+				.perform(click());
+		try {
```

Comment:
i really dislike try / catches in test code.
also this method leaves the user/tester in a different fragment than where he was when he started it.

Reply:
I don't like try&catch eighter but I have no idea how I am supposed to do it without. In my last pull request, you told me that you want me to make a function of it which can also be used in another test case. However, depending on if that variable is in use there is an additional Dialog. Do you have an Idea how to check if the deleted variable is in use by the user/tester to make a function for this without try/catch?

Reply:
yes, you can get the variable list from the project and check if it is in there...
or you can have the dev that uses this function take care of this by adding a parameter to the function.